### ISSUE TITLE
Math::BigInt::Pari backend doesn't work with Math::BigRat

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8732]
Migrated from [rt.perl.org#41205](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=41205) (status was 'resolved')

Searchable as RT41205$

**Comment by p5pRT** [comment]
### From zefram@fysh.org

#### Created by zefram@fysh.org



$ perl \-MMath&#8203;::BigRat=lib\,Pari \-lwe 'Math&#8203;::BigRat\->new\("7/3"\)'
n is not Math&#8203;::BigInt&#8203;::Pari but \(Math&#8203;::Pari\) in bnorm\(\) at \-e line 1
$

The Pari backend manages to do BigInt stuff fine\.  It's only with BigRat
that it fails\.  Version numbers&#8203;:

Math&#8203;::BigInt 1\.77
Math&#8203;::BigRat 0\.15
Math&#8203;::BigInt&#8203;::Pari 1\.11
Math&#8203;::Pari 2\.010709

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=high

Site configuration information for perl v5.8.4:

Configured by Debian Project at Wed May 10 04:14:05 UTC 2006.

Summary of my perl5 (revision 5 version 8 subversion 4) configuration:
  Platform:
    osname=linux, osvers=2.6.15.6, archname=i386-linux-thread-multi
    uname='linux ernie 2.6.15.6 #1 thu mar 16 13:11:55 est 2006 i686 gnulinux '
    config_args='-Dusethreads -Duselargefiles -Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=i386-linux -Dprefix=/usr -Dprivlib=/usr/share/perl/5.8 -Darchlib=/usr/lib/perl/5.8 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.8.4 -Dsitearch=/usr/local/lib/perl/5.8.4 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Uusesfio -Uusenm -Duseshrplib -Dlibperl=libperl.so.5.8.4 -Dd_dosuid -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -DDEBIAN -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -DDEBIAN -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='3.3.5 (Debian 1:3.3.5-13)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=/lib/libc-2.3.2.so, so=so, useshrplib=true, libperl=libperl.so.5.8.4
    gnulibc_version='2.3.2'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.4:
    /etc/perl
    /usr/local/lib/perl/5.8.4
    /usr/local/share/perl/5.8.4
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.8
    /usr/share/perl/5.8
    /usr/local/lib/site_perl
    .


Environment for perl v5.8.4:
    HOME=/home/zefram
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/zefram/pub/i686-pc-linux-gnu/bin:/home/zefram/pub/common/bin:/usr/bin:/usr/X11R6/bin:/bin:/usr/local/bin:/usr/games
    PERL_BADLANG (unset)
    SHELL=/usr/bin/zsh

```
</details>

**Comment by p5pRT** [comment]
### From nospam-abuse@bloodgate.com

\-\-\-\-\-BEGIN PGP SIGNED MESSAGE\-\-\-\-\-
Hash&#8203;: SHA1

Moin\,

this issue is tracked under this bug \#25276 here&#8203;:

&nbsp;    http&#8203;://rt\.cpan\.org/Ticket/Display\.html?id=25276

Sorry for the delay\,

Tels

\- \-\- 
 Signed on Mon Apr 16 16&#8203;:24&#8203;:04 2007 with key 0x93B84C15\.
 View my photo gallery&#8203;: http&#8203;://bloodgate\.com/photos
 PGP key on http&#8203;://bloodgate\.com/tels\.asc or per email\.

 "Now\, \_you\_ behave\!"

\-\-\-\-\-BEGIN PGP SIGNATURE\-\-\-\-\-
Version&#8203;: GnuPG v1\.4\.2 \(GNU/Linux\)

iQEVAwUBRiOjZXcLPEOTuEwVAQIEQwf/VjIHFVZ0fvi3a1kMlDTIUV8CD1tuXi1Y
Mb9LI/upGWuCQdRQtZREDXveMicKmhUNa9Omw\+rlbxTNw4gYMSYsKB/IacyE3fes
JALVVJqK3Lg\+GB4BaRhSbpr4A5tsCNbINfwuVHJ5FZg1deYv0kTm\+UfeVeYtCmgo
9gB4sF\+SeCB9POQP9LArf2i4Wf6KUY2jQ6\+YxsR\+50zVgeYaTRgwG1suBKKGkWAG
VYhUsmzut596JXV80Wy70ibqKcs1mF7J2JJ8dQuVfnXTa1qxSsywQYxzCNsms\+qy
\+oM8rzqPyy7H84ukEAR35X0/oT40OIudY/e272AVA74TK4ieMk0zPA==
=exza
\-\-\-\-\-END PGP SIGNATURE\-\-\-\-\-

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From nospam-abuse@bloodgate.com

\-\-\-\-\-BEGIN PGP SIGNED MESSAGE\-\-\-\-\-
Hash&#8203;: SHA1

Moin\,

On Monday 16 April 2007 16&#8203;:25&#8203;:09 Tels wrote&#8203;:
> Moin\,
>
> this issue is tracked under this bug \#25276 here&#8203;:
>
>     http&#8203;://rt\.cpan\.org/Ticket/Display\.html?id=25276
>
> Sorry for the delay\,
>
> Tels

This has been fixed with the release of Math&#8203;::BigRat v0\.19&#8203;:

&nbsp;    http&#8203;://search\.cpan\.org/~tels/Math\-BigRat/

Could somebody please close this Perl bug?

Thanx\,

Tels

\- \-\- 
 Signed on Wed Apr 18 12&#8203;:07&#8203;:05 2007 with key 0x93B84C15\.
 Get one of my photo posters&#8203;: http&#8203;://bloodgate\.com/posters
 PGP key on http&#8203;://bloodgate\.com/tels\.asc or per email\.

 "Blogebrity&#8203;: Wow\, guess what this one stands for? Too easy\. Hey\, anyone
 can do it&#8203;: take a blogger who's a chef\, and you get&#8203;: BLEF\. A blogger
 who's a dentist? BENTIST\. A female blogger with an itch? You guessed it&#8203;:
 a BITCH\."

&nbsp;  \-\- maddox from xmission
\-\-\-\-\-BEGIN PGP SIGNATURE\-\-\-\-\-
Version&#8203;: GnuPG v1\.4\.2 \(GNU/Linux\)

iQEVAwUBRiYKH3cLPEOTuEwVAQLX1gf\+MgweoO9Zp7Ju8J9LC54eS4vauf4RcOvy
fqJ1jnv3EHa3UNFBq3E11OHlcjwk30D3TbCtZxW7fpii/pa8K18bK8HTs4ysd6FY
n6hDVDDkxTqdjUe0Ck5WyU7Q/UXTsEcBn9vCCRXYlbhsJ75yLNsIvyIdNb2rjKwJ
0csqFIbveAGyMhJSGDb3ocb7edSPm04UvN167yPDHhjcacrfNXugdfhzhsxOFqWF
KX1qFBKMgVjrWuV4YOdFt5Y6nH2jKY3hZBz1C3eX2HZWt0AbQOGWB6QnKSBz7Dpd
d8U9dB451qnw8\+r5lnjjUzfqM7ubAnxoS4GEpChZv2W7soY4Hqryjg==
=rs\+P
\-\-\-\-\-END PGP SIGNATURE\-\-\-\-\-
